dev/src/write-batch.ts

```diff
@@ -577,7 +586,54 @@ export class WriteBatch implements firestore.WriteBatch {
       api.BatchWriteResponse
     >('batchWrite', request, tag, retryCodes);
 
-    return (response.writeResults || []).map((result, i) => {
+    let retryCount = 0;
+    while (retryCount < MAX_BATCH_WRITE_RETRY_ATTEMPTS) {
+      // Find the indexes of all writes that failed with ABORTED.
+      const abortedIndexes = response.status.reduce(
```

Comment:
Let's make this generic and base our retries on the behavior defined in https://github.com/googleapis/nodejs-firestore/blob/master/dev/src/v1/firestore_client_config.json

This is a multi-step process though: First, we need to modify the `firestore_client_config.json` in Google3. We likely want to retry writes on both UNAVAILABLE and ABORTED. We can then use https://github.com/googleapis/nodejs-firestore/blob/f175236bde2f64365f140b14641f848bd4eb34d9/dev/src/util.ts#L165 to extract the retry codes.